helper.py

```diff
@@ -32,9 +45,91 @@ def get_req_handler(url: str) -> requests.Response:
     return response
 
 
-# Initialize boto3 session from env vars (ensure load_dotenv fn is called beforehand)
-def initBoto3Session() -> None:
-    boto3.Session(
-        aws_access_key_id=os.environ["AWS_ACCESS_KEY_ID"],
-        aws_secret_access_key=os.environ["AWS_SECRET_ACCESS_KEY"],
+# returns string form of datetime timestamp (from API calls)
+def formatted_timestamp(timestamp):
+    return dateutil.parser.parse(timestamp).strftime("%Y-%m-%dT%H%M%S")
+
+
+# dataframe: pandas dataframe to save as CSV (locally or in AWS bucket)
+# directory: folder to save the CSV file
+# filename: name of the CSV file (ending with .csv)
+# output_loc: one of 'local', 'AWS'
+# s3_bucket: S3.Bucket object where the CSV file is to be pushed
+def output_csv(
+    dataframe: pd.DataFrame,
+    directory: str,
+    filename: str,
+    output_loc: str = "local",
+    s3_bucket: Bucket = None,
+):
+    savepath = directory + filename
+    if output_loc == "local":
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+        dataframe.to_csv(savepath, encoding="utf-8")
+    elif output_loc == "AWS":
+        if s3_bucket:
+            try:
+                # upload dataframe to S3 bucket
+                csv_buffer = StringIO()
+                dataframe.to_csv(csv_buffer, encoding="utf-8")
+                s3_bucket.put_object(Body=csv_buffer.getvalue(), Key=savepath)
+            except Exception:
+                print("Failed to upload data in " + savepath, file=sys.stderr)
+            else:
+                print("Uploaded dataframe as csv file " + savepath + " to bucket!")
```

Comment:
Shortening this portion with the same effect
```suggestion
                s3_bucket.put_object(Body=csv_buffer.getvalue(), Key=savepath)
                print("Uploaded dataframe as csv file " + savepath + " to bucket!")
            except Exception:
                print("Failed to upload data in " + savepath, file=sys.stderr)
```